### PR TITLE
docs: add wiki FAQ to the runtime documentation

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: runtime/doc/faq.txt
Comment (lucario387) [inline runtime/doc/faq.txt:188]: shouldn't this be `>bash`

Comment (clason) [reply runtime/doc/faq.txt:188]: Yes, I plan to go through the document and fix things (and remove outdated information) sometime today.

File: runtime/doc/faq.txt
Comment (clason) [inline runtime/doc/faq.txt:447]: That whole section probably belongs in CONTRIBUTING.md or MAINTAINING.md (or a new `:h dev_tools` file or somesuch); it's a bit out of place here.

File: scripts/gen_help_html.lua
Comment (justinmk) [inline scripts/gen_help_html.lua:84]: https seems to work for this one: https://seriot.ch/projects/parsing_json.html

Comment (dundargoc) [reply scripts/gen_help_html.lua:84]: I didn't write url checkeer

Comment (dundargoc) [reply scripts/gen_help_html.lua:84]: All of them work

Comment (clason) [reply scripts/gen_help_html.lua:84]: Probably worth a pass across the docs to switch to https where possible (separate PR).

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:4]: Type |gO| to see the table of contents.

**Comment by clason** [comment]
@dundargoc you force-pushed over my changes. What did you edit?

**Comment by dundargoc** [comment]
Sus. I think I just rebased.

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:9]: I think we can fix the parser so that UPPERCASE lines support punctuation, then the trailing `~` won't be needed. This is fine for now though.

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:138]: assuming we still want this faq item, we can just link to `|clipboard-wsl|` and don't need to repeat the code sample here.

Comment (clason) [reply runtime/doc/faq.txt:138]: If in doubt, remove. I've done so for obviously outdated content but erred on the side of caution.

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:8]: I think we should not number things manually. Let `gO` do that, or just live without it.
```suggestion
General Questions                                 *faq* *faq-general-questions*
```

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:240]: should probably move this note to `:help E518`

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:285]:
```suggestion
NOTE: Nvim ignores `t_Co` and other |t_xx| terminal codes.
```

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:389]: this could go under "build issues"

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt]:
```suggestion
1. General Questions                                 *faq* *faq-general*
```

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt]:
```suggestion
2. Runtime issues                                         *faq-runtime*
```

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt]:
```suggestion
4. Build issues                                             *faq-build*
```

File: runtime/doc/faq.txt
Comment (clason) [reply runtime/doc/faq.txt:11]: (also modelines)

File: runtime/doc/faq.txt
Comment (justinmk) [reply runtime/doc/faq.txt]: `dev_tools.txt` sounds good

File: runtime/doc/help.txt
Comment (justinmk) [inline runtime/doc/help.txt:50]:
```suggestion
|faq|			Frequently Asked Questions
```

File: runtime/doc/faq.txt
Comment (justinmk) [inline runtime/doc/faq.txt:645]: the redis post is good, let's keep it

Comment (clason) [reply runtime/doc/faq.txt:645]: Really? It doesn't really mention Lua in any detail. What's good about it?

Comment (justinmk) [reply runtime/doc/faq.txt:628]: similar architecture as Nvim, similar strategy for batching work and avoiding a big RPC API

**Review by justinmk — APPROVED** [review]
LGTM after dev_tools.txt.

Thanks for doing this. "Single source of truth" is high-leverage for users + maintainers.

The old faq wiki page should point to the new page, since it is the target of many old links.

**Comment by clason** [comment]
We'll make a followup PR to neovim.io once we're done cleaning out the Augean Wiki.

**Comment by justinmk** [comment]
> Nah, let's remove it. Let's fix the old links instead.

Yes, but I think many things we don't control link to the faq. E.g. old reddit posts, articles, etc. The FAQ is common enough that keeping a handle around will be helpful. For other less common wiki pages we can just nuke them.

**Comment by dundargoc** [comment]
Aight, let's keep it for now then 👍

**Comment by clason** [comment]
I think this is ready. Please don't squash-merge; either rebase or make a merge commit to preserve the treesitter fix.

File: runtime/doc/dev_tools.txt
Comment (justinmk) [inline runtime/doc/dev_tools.txt:7]:
```suggestion
Nvim development tooling                                        *dev-tools*
```

File: runtime/doc/dev_tools.txt
Comment (justinmk) [inline runtime/doc/dev_tools.txt:10]: Can revisit later, but just FYI: I see this document as a place where we can document more than just debugger stuff.

**Review by justinmk — APPROVED** [review]

File: runtime/doc/dev_tools.txt
Comment (clason) [reply runtime/doc/dev_tools.txt:10]: Such as? (I'd be happy to generalize this already if I have some idea.)

File: runtime/doc/dev_tools.txt
Comment (clason) [reply runtime/doc/dev_tools.txt:7]: whoops...

File: runtime/doc/dev_tools.txt
Comment (justinmk) [reply runtime/doc/dev_tools.txt]: Perhaps some minor hints about development setup / LSP ?, Some parts of `src/nvim/README.md`, such as ASAN, getting a stacktrace, etc.

For codebase-related info (data structures, usage of various modules, etc.) we'll probably need a separate `dev_xx.txt` file, or we could put that in here too.

Comment (clason) [reply runtime/doc/dev_tools.txt]: Agreed; that is the goal. I've left it at a vague "working on Nvim and debugging issues" for now (and updated the description in `:h help`).

**Pull request merged**
